### PR TITLE
Monit 26299 golang-otel-example

### PR DESCRIPTION
**Pull request by deosu**

### CODE REVIEW

File: golang-example/main.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"log"
+	"time"
+)
+
+func initTracer() func() {
+
+	ctx := context.Background()
```
Comment (thepeterstone) [inline golang-example/main.go:22]: Might be tidier to do 
```
ctx, cancel := context.WithCancel(context.Background())
```
and call `cancel` in the shutdown function.

**Review by thepeterstone — CHANGES_REQUESTED** [review]
A few minor changes and typos - otherwise LGTM!

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -96,7 +96,7 @@ jobs:
         run: |
           dotnet run --project DOTNET-example/Web-App &
           TEMP_OUTPUT_FILE=$(mktemp)
-          WAIT_TIME=15
+          WAIT_TIME=20
```
[the remark this answers is not in the document]
Comment (deosu) [reply .github/workflows/main.yml:99]: Ideally, 15 seconds should be enough, but sometimes I have noticed it takes more than that. As we have added more jobs that run in parallel, I thought it is better to give 5 seconds more to finish a job.

File: .gitignore [hunk 3]
```diff
@@ -56,7 +56,6 @@ buildNumber.properties
 *.dll
 
 # Fortran module files
-*.mod
```
[the remark this answers is not in the document]
Comment (deosu) [reply .gitignore:59]: Sure, I'll add that.

File: go-example/auto-instrumentation/README.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+* Install the Tanzu Observability proxy. See
+  this [README](https://github.com/wavefrontHQ/opentelemetry-examples/blob/master/README.md#install-wavefront-proxy).
+* Set up an OpenTelemetry Collector for Tanzu Observability. See
+  this [README](https://github.com/wavefrontHQ/opentelemetry-examples/blob/master/README.md#install-the-opentelemetry-collector)
+  .
+
+### Step 1: Install OpenTelemetry Packages
+
+All the required dependencies are listed
+in [`mod.go`](https://github.com/wavefrontHQ/opentelemetry-examples/blob/master/go-example/go.mod). To install
```
[the remark this answers is not in the document]
Comment (deosu) [reply go-example/auto-instrumentation/README.md:20]: Resolved.

File: go-example/auto-instrumentation/README.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
```
[the remark this answers is not in the document]
Comment (deosu) [reply go-example/auto-instrumentation/README.md:98]: Resolved.

File: go-example/auto-instrumentation/main.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+	}
+	resources, err := resource.New(
+		context.Background(),
+		resource.WithAttributes(
+			attribute.String("service.name", "otel-otlp-go-service"),
+			attribute.String("application", "otel-otlp-go-app"),
+		),
+	)
+	if err != nil {
+		log.Printf("Could not set resources: ", err)
```
[the remark this answers is not in the document]
Comment (deosu) [reply go-example/auto-instrumentation/main.go:38]: It should be Fatal, we can not continue without resources.

File: go-example/manual-instrumentation/README.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+* Install the Tanzu Observability proxy. See
+  this [README](https://github.com/wavefrontHQ/opentelemetry-examples/blob/master/README.md#install-wavefront-proxy).
+* Set up an OpenTelemetry Collector for Tanzu Observability. See
+  this [README](https://github.com/wavefrontHQ/opentelemetry-examples/blob/master/README.md#install-the-opentelemetry-collector)
+  .
+
+### Step 1: Install OpenTelemetry Packages
+
+All the required dependencies are listed
+in [`mod.go`](https://github.com/wavefrontHQ/opentelemetry-examples/blob/master/go-example/go.mod).
```
[the remark this answers is not in the document]
Comment (deosu) [reply go-example/manual-instrumentation/README.md:20]: Resolved.

File: go-example/manual-instrumentation/main.go [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+			semconv.ServiceNameKey.String("otel-otlp-go-service"),
+			attribute.String("application", "otel-otlp-go-app"),
+		),
+	)
+	return res, err
+}
+
+func handleErr(err error, message string) {
+	if err != nil {
+		log.Printf("%s: %v", message, err)
```
[the remark this answers is not in the document]
Comment (deosu) [reply go-example/manual-instrumentation/main.go:74]: agree, fixed.

**Comment by deosu** [comment]
@thepeterstone can you please review again and approve?

**Review by thepeterstone — APPROVED** [review]

**Pull request merged**